Source/WebCore/dom/Document.cpp

```diff
@@ -10772,6 +10773,31 @@ HTMLElement* Document::topmostAutoPopover() const
     return m_autoPopoverList.last().ptr();
 }
 
+HTMLDialogElement* Document::nearestClickedDialog(const PointerEvent& event, Node& target) const
+{
+    if (RefPtr dialog = dynamicDowncast<HTMLDialogElement>(target)) {
+        Ref<DOMRect> dialogRect = dialog->getBoundingClientRect();
+        auto insideBounds = event.clientX() >= dialogRect->left()
+            && event.clientX() < dialogRect->right()
+            && event.clientY() >= dialogRect->top()
+            && event.clientY() < dialogRect->bottom();
+        if (dialog->isOpen() && dialog->isModal() && !insideBounds)
+            return nullptr;
+    }
```

Comment:
The HitTestRequest that populates the event target should have data on whether it's the backdrop or the dialog being clicked. I think it would be reasonable to extend the relevant Event interface with a non-web exposed field to pass on that data here

Reply:
https://bugs.webkit.org/show_bug.cgi?id=298200 - it's not immediately clear to me how to trace the exact course of this PointerEvent, I've raised a follow-up issue to look into this.